test/db/cmd/midflags

```diff
@@ -395,3 +395,25 @@ EXPECT=<<EOF
   0x00000009      90             nop
 EOF
 RUN
+
+NAME=asm.flags.middle reset
+FILE=-
+CMDS=<<EOF
+e asm.arch=x86
+e asm.bytes=true
+e asm.bits=64
+e asm.bb.middle=false
+e asm.flags.middle=2
+wx b821c10010ebf9b8210011c1
+af @ 3
+f sym.dummy @ 9
+pd 4 @ 0
+EOF
+EXPECT=<<EOF
+|           0x00000000  ~   b821c10010     mov   eax, 0x1000c121
```

Comment:
Why is this better than the current output:
```
[0x00000000]> pd 4 @ 0
│           ; CODE XREF from fcn.00000003 @ 0x5
│           0x00000000  ~   b821c10010     mov   eax, 0x1000c121
┌ 9: fcn.00000003 ();
│ bp: 0 (vars 0, args 0)
│ sp: 0 (vars 0, args 0)
│ rg: 0 (vars 0, args 0)
│           0x00000003      0010           add   byte [rax], dl
│           0x00000005      ebf9           jmp   0
            0x00000007  ~   b8210011c1     mov   eax, 0xc1110021       ; '!'
```
??

Reply:
Because `asm.flags.middle=2` aka realign on flag and there is a flag at address 9.

Reply:
Isn't there a flag also at address 3? In particular, the user has even said there is a function starting at 3 and the new output in your PR hides that.

Reply:
It's #664 that hides the function at 3. Why? Because realigning on 3 breaks graph disassembly output (see #664's test) by showing an instruction that is not executed by the CPU.

Due to the limitations of a linear disassembly view, you can still realign on 3 by setting `asm.bb.middle=true`.

Reply:
You might argue that graph view should set `asm.flags.middle=1`. But then we can get into a situation where the linear view realigns at 3 regardless of the value of `asm.bb.middle`.

Imo, flags should not be able to break up the instruction flow in a basic block. Perhaps the only reason you want to do that is if the instructions are actually hiding something else, but right now (if I'm reading the code correctly), both the linear and graph disassembly will realign on a `str.` flag regardless.

That the flag at 3 is hidden unfortunately appears to be a bug.

Reply:
> ... both the linear and graph disassembly will realign on a `str.` flag regardless.

Except on `asm.flags.middle=0` that is.

Reply:
:/ you lost me! 

Who said what is executing and what is not? I mean, are you assuming that execution starts at 0?

I guess that in linear view there is no 100% perfect way to show these kind of situations, right?

Reply:
> :/ you lost me!

Sorry.

> Who said what is executing and what is not? I mean, are you assuming that execution starts at 0?

The test does an `af @ 3`. There is a basic block starting from 0x0.

> I guess that in linear view there is no 100% perfect way to show these kind of situations, right?

Yes, but you can make it easy to toggle between views.

Reply:
> > Who said what is executing and what is not? I mean, are you assuming that execution starts at 0?
> 
> The test does an `af @ 3`. There is a basic block starting from 0x0.

Right, so essentially with this new PR you preferred the bb at 0x0 instead of the one at 0x3, right? Why should it be better to show the one at 0 rather than the one at 3? Maybe We should just find a better way to clearly show both paths? Just saying, i'm not sure yet how.

Reply:
> Right, so essentially with this new PR you preferred the bb at 0x0 instead of the one at 0x3, right? Why should it be better to show the one at 0 rather than the one at 3?

You can show the one at 3. Just set `asm.bb.middle=true` and by the way, this is the default value of `asm.bb.middle`. This PR has nothing to do with whether 0 or 3 is preferred.

> Maybe We should just find a better way to clearly show both paths?

Just do `agf` or `VV`.

Reply:
Ok!

Reply:
Tq! :tada::scream::joy::dancer: